native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/Engine.java

```diff
@@ -236,4 +242,22 @@ public static BString getInterceptorName(BObject interceptor) {
         ServiceType serviceType = (ServiceType) interceptor.getType();
         return StringUtils.fromString(serviceType.getName());
     }
+
+    public static Object getResourceAnnotation(BObject service, BString operationType, BArray path,
+                                               BString methodName) {
+        ServiceType serviceType = (ServiceType) service.getType();
+        BString identifier = StringUtils.fromString(getModule().toString() + COLON + RESOURCE_CONFIG);
+        MethodType methodType = null;
+        if (OPERATION_QUERY.equals(operationType.getValue())) {
+            methodType = getResourceMethod(serviceType, getPathList(path), GET_ACCESSOR);
+        } else if (OPERATION_SUBSCRIPTION.equals(operationType.getValue())) {
+            methodType = getResourceMethod(serviceType, getPathList(path), SUBSCRIBE_ACCESSOR);
+        } else {
+            methodType = getRemoteMethod(serviceType, String.valueOf(methodName));
+        }
+        if (methodType != null) {
+            return methodType.getAnnotation(identifier);
```

Comment:
Since we are only using the variable `identifier` here we could move declaration inside if condition